code/modules/mob/living/silicon/silicon.dm

```diff
@@ -353,3 +354,26 @@
 	..()
 	if(cameraFollow)
 		cameraFollow = null
+
+/mob/living/silicon/Move(newloc, direct)
+	..(newloc,direct)
+	if (underdoor)
+		if (!(layer == UNDERDOOR))

```

Comment:
If you're forcing `underdoor = 0` regardless of circumstance on line line 363, then why do it conditionally just beforehand? Mistake, or oversight?